cmake/dependencies.cmake

```diff
@@ -56,3 +56,38 @@ CPMAddPackage(
   GITHUB_REPOSITORY boost-ext/reflect
   GIT_TAG v1.1.1
 )
+
+############################################################################################################################
+# Packages needed for tt-metal simulator
+#   NNG for IPC/TCP communication
+#   Google Flatbuffers for serialization
+#   libuv for process mgmt
+############################################################################################################################
+CPMAddPackage(
```

Comment:
if we switched back to a UMD library, we wouldn't need these here right? They're already declared with your UMD changes

Reply:
Yes, I think if we switched back to `add_subdirectory` or `ExternalProject_Add` for umd, we shouldn't need to bring these in at metal level.